src/DataTypes/convertMySQLDataType.cpp

```diff
@@ -73,7 +73,7 @@ DataTypePtr convertMySQLDataType(MultiEnum<MySQLDataTypesSupport> type_support,
     else if (type_name == "double")
         res = std::make_shared<DataTypeFloat64>();
     else if (type_name == "date")
-        res = std::make_shared<DataTypeDate>();
+        res = std::make_shared<DataTypeDate32>();
```

Comment:
Let's make this under a setting `convert_mysql_date_to_date32`

Reply:
done, I have added DATE2DATE32 and DATE2STRING to MySQLDataTypesSupport.